woocommerce/woocommerce
# Set stock quantity value as 0 by default

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR reverts the changes added in https://github.com/woocommerce/woocommerce/issues/37117 and defaults the stock quantity to 0.

We used to default the stock quantity to 1 when a product with enabled stock management was added without specifying a stock value. However, it was `null` in other areas of the system. This inconsistency was problematic and led to unexpected behaviors.

The changes added here:
- Set the stock quantity to 0 by default.
- It's not possible [to set the stock quantity to `null`](https://github.com/woocommerce/woocommerce/pull/48448/files#diff-d62350bd045581ade84f14b7a7600040162cc905417e817a9cae27cfd3ac93aeR969) anymore
- `get_stock_quantity` doesn't return `null` anymore. It will return an integer. [#](https://github.com/woocommerce/woocommerce/pull/48448/files#diff-ef27d662bcec590c665fc7ac8c4a698c5a0a40af67376751fa2299c2541c0a79R341) and [#](https://github.com/woocommerce/woocommerce/pull/48448/files#diff-d62350bd045581ade84f14b7a7600040162cc905417e817a9cae27cfd3ac93aeR364)

For additional details https://github.com/woocommerce/woocommerce/issues/39475

![Screenshot 2024-06-14 at 8 54 57 AM](https://github.com/woocommerce/woocommerce/assets/1314156/15207805-cbf6-4885-bb9c-071a5b24d64a)

and

![Screenshot 2024-06-14 at 8 55 51 AM](https://github.com/woocommerce/woocommerce/assets/1314156/1a2a2e9d-6402-46a9-97c6-2eb4b491b4be)

Closes #46342.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch.
2. Go to Products > Add New > Inventory.
3. Check `Track stock quantity for this product`.
4. Verify that `Quantity` is 0.
5. Create a product and confirm that the stock management works well.
6. Try also a few extensions that interacts with the stock quantity like `WooCommerce Product Bundles` and `WooCommerce Back in Stock Notifications`.
7.  Make some smoke tests.
8. Make sure that `Try the new product editor (Beta)` is enabled under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
9. Go to Products > Add New > Inventory.
10. Press `Track inventory`.
11. Verify the `Available stock` is 0.
12. Make some smoke tests.
13. Test with variable products as well.




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
